pallets/msa/src/lib.rs

```diff
@@ -112,13 +130,48 @@ pub mod pallet {
 		pub fn create(origin: OriginFor<T>) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
-			let (identifier, _) = Self::create_account(who.clone())?;
+			let (identifier, _) = Self::create_account(who.clone(), EMPTY_FUNCTION)?;
 
 			Self::deposit_event(Event::MsaCreated { msa_id: identifier, key: who });
 
 			Ok(())
 		}
 
+		#[pallet::weight(T::WeightInfo::create(10_000))]
+		pub fn create_account_with_delegate(
+			origin: OriginFor<T>,
+			proof: MultiSignature,
+			creation_delegate_payload: AccountCreationDelegate<T::AccountId>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			Self::verify_signature(
+				proof,
+				creation_delegate_payload.key.clone(),
+				creation_delegate_payload.encode(),
+			)?;
+
```

Comment:
we need add in here that they key is valid
```
Self::ensure_valid_msa_key(&delegate_key (submitter))?
I think we want to check that the signing key of the transaction is valid.
```

Reply:
we have this inside `ensure_valid_creation_delegate`

Reply:
ensure_valid_creation_delegate does not really do anything? Lets jump on a call.